packages/editor/src/components/provider/index.native.js

```diff
@@ -232,15 +228,23 @@ class NativeEditorProvider extends Component {
 		}
 	}
 
-	getThemeColors( { colors, gradients, rawStyles, rawFeatures } ) {
-		return {
-			...( rawStyles && rawFeatures
-				? getGlobalStyles( rawStyles, rawFeatures )
-				: {
-						colors: validateThemeColors( colors ),
-						gradients: validateThemeGradients( gradients ),
-				  } ),
-		};
+	getThemeColors( { rawStyles, rawFeatures } ) {
+		const { defaultEditorColors, defaultEditorGradients } = this.props;
+
+		if ( rawStyles && rawFeatures ) {
+			return getGlobalStyles( rawStyles, rawFeatures );
+		}
+
+		if (
+			( ! rawStyles && rawFeatures ) ||
+			( ! rawStyles && ! rawFeatures )
```

Comment:
Thanks for spotting this, I ended up [removing this check](https://github.com/WordPress/gutenberg/pull/38417/commits/9eb0f97ea0c37b3cdda39941081eacc92e5fae15), if it doesn't have `rawStyles` and `rawFeatures` it will always return `getColorsAndGradients` which internally checks if `rawFeatures` exists.